pkg/server/endpoints/authorized_entryfetcher_test.go

```diff
@@ -190,6 +191,50 @@ func TestBuildRegistrationEntriesCache(t *testing.T) {
 	}
 }
 
+func TestBuildCacheSavesMissedEvents(t *testing.T) {
+	ctx := context.Background()
+	log, _ := test.NewNullLogger()
+	clk := clock.NewMock(t)
+	ds := fakedatastore.New(t)
+
+	// Create Registration Entry Events with a gap
+	_, err := ds.CreateRegistrationEntryEvent(ctx, &datastore.RegistrationEntryEvent{
+		EventID: 1,
+		EntryID: "test",
+	})
+	require.NoError(t, err)
+
+	_, err = ds.CreateRegistrationEntryEvent(ctx, &datastore.RegistrationEntryEvent{
+		EventID: 3,
+		EntryID: "test",
+	})
+	require.NoError(t, err)
+
+	// Create AttestedNode Events with a gap
+	_, err = ds.CreateAttestedNodeEvent(ctx, &datastore.AttestedNodeEvent{
+		EventID:  1,
+		SpiffeID: "test",
+	})
+	require.NoError(t, err)
+
+	_, err = ds.CreateAttestedNodeEvent(ctx, &datastore.AttestedNodeEvent{
+		EventID:  4,
+		SpiffeID: "test",
+	})
+	require.NoError(t, err)
+
+	ef, err := NewAuthorizedEntryFetcherWithEventsBasedCache(ctx, log, clk, ds, defaultCacheReloadInterval, defaultPruneEventsOlderThan)
+	require.NoError(t, err)
+	require.NotNil(t, ef)
+
+	assert.Contains(t, ef.missedRegistrationEntryEvents, uint(2))
+	assert.Equal(t, uint(3), ef.lastRegistrationEntryEventID)
+
+	assert.Contains(t, ef.missedAttestedNodeEvents, uint(2))
+	assert.Contains(t, ef.missedAttestedNodeEvents, uint(3))
```

Comment:
Faisal checked for equality of the count of returned items on line 231.  Note that the actual items can come in any order (due to SQL and not having an order statement here) so it is correct to check each item is contained in the collection instead of assuming some sort of index based equality.

Reply:
it is a map... so order does not matter when comparing maps, with that equal will guaranty that you have 2 entries and the expected entries

Reply:
Would you suggest the alternative approach you are seeking here?

Reply:
And yeah, what marcos said works. Map equality checks are already order independent.